src/test/run-pass-fulldeps/proc-macro/auxiliary/derive-union.rs

```diff
@@ -0,0 +1,30 @@
+// Copyright 2018 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
+// http://rust-lang.org/COPYRIGHT.
+//
+// Licensed under the Apache License, Version 2.0 <LICENSE-APACHE or
+// http://www.apache.org/licenses/LICENSE-2.0> or the MIT license
+// <LICENSE-MIT or http://opensource.org/licenses/MIT>, at your
+// option. This file may not be copied, modified, or distributed
+// except according to those terms.
+
+// no-prefer-dynamic
+
+#![crate_type = "proc-macro"]
+
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+
+const EXPECTED: &'static str =
+"#[repr(C)]
+union Test {
+    a: u8,
+}";
+
+#[proc_macro_derive(Union)]
+pub fn derive(input: TokenStream) -> TokenStream {
+    let input = input.to_string();
+    assert_eq!(EXPECTED, input);
```

Comment:
Other derive tests appear to just assert that the input contains certain lines: https://github.com/rust-lang/rust/blob/master/src/test/run-pass-fulldeps/proc-macro/auxiliary/derive-b.rs#L23

I don't think it affects the confidence interval of the test much since we're mostly just ensuring that derives are allowed to be applied to unions. I'll let @alexcrichton make the call on what he prefers to see.